### PR TITLE
Remove phrase about kernel argument pointer types

### PR DESCRIPTION
**Pull request by svenvh**
This phrase seems to be purely advisory and is not an explicit
restriction.  The Clang implementation has always been accepting
pointer types for the affected types.

Fixes https://github.com/KhronosGroup/OpenCL-Docs/issues/504

Signed-off-by: Sven van Haastregt <sven.vanhaastregt@arm.com>

### CODE REVIEW

**Review by bashbaug — APPROVED** [review]
This makes sense and looks good to me.  I'd like to wait for working group signoff (or at least another reviewer) before merging, though.

**Review by alycm — APPROVED** [review]
LGTM, I agree that this sentence doesn't add anything concrete to the previous one.

**Pull request merged**
